### ISSUE TITLE
The parameters 'mainClass' for goal io.github.fvarrui:javapackager:1.6.7:package are missing or invalid

### ISSUE THREAD

**Issue by guohao0328** [fvarrui/JavaPackager#280]
JavaFX项目

操作系统 macOS 10.14.6

jdk版本：11.0.16

开发工具：IDEA

构建工具：maven

javapackage版本：1.6.7

双击右侧maven plugin   javapackage:package

报错
The parameters 'mainClass' for goal io.github.fvarrui:javapackager:1.6.7:package are missing or invalid

```
      <plugin>
        <groupId>io.github.fvarrui</groupId>
        <artifactId>javapackager</artifactId>
        <version>1.6.7</version>
        <executions>
          <execution>
            <phase>package</phase>
            <goals>
              <goal>package</goal>
            </goals>
            <configuration>
              <!-- mandatory -->
              <mainClass>org.guohao.mockdata.MockDataApplication</mainClass>
              <!-- optional -->
              <bundleJre>true</bundleJre>
              <generateInstaller>false</generateInstaller>
              <administratorRequired>false</administratorRequired>
              <!--              auto|linux|mac|windows-->
              <platform>mac</platform>
              <!--              <additionalResources>-->
              <!--                <additionalResource>file path</additionalResource>-->
              <!--                <additionalResource>folder path</additionalResource>-->
              <!--                <additionalResource>...</additionalResource>-->
              <!--              </additionalResources>-->
              <!--              <linuxConfig>...</linuxConfig>-->
              <!--              <macConfig>...</macConfig>-->
              <!--              <winConfig>...</winConfig>-->
              <!--              [...]-->
            </configuration>
          </execution>
        </executions>
      </plugin>
```

main class
```
package org.guohao.mockdata;

import java.io.IOException;
import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.geometry.Rectangle2D;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Screen;
import javafx.stage.Stage;
import org.guohao.mockdata.controller.HomeController;

public class MockDataApplication extends Application {

  @Override
  public void start(Stage stage) throws IOException {
    FXMLLoader fxmlLoader = new FXMLLoader(
        MockDataApplication.class.getResource("/fxml/home.fxml"));
    Rectangle2D screenBounds = Screen.getPrimary().getBounds();
    Scene scene = new Scene((Parent) fxmlLoader.load(), screenBounds.getWidth() * 0.6,
        screenBounds.getHeight() * 0.6);
    HomeController controller = fxmlLoader.getController();
    controller.mainStage = stage;
    stage.setTitle("Hello!");
    stage.setScene(scene);
    stage.show();
  }

  public static void main(String[] args) {
    launch();
  }
}
```

**Comment by fvarrui** [comment]
Hi @guohao0328!
There's a known bug in v.1.6.7 on Mac OS that will be fixed soon.
Could you try with 1.6.6, please? Or 1.7.0-SNAPSHOT from `devel` branch?

**Comment by fvarrui** [comment]
You can also share your sample project, so I can try it by myself.

**Comment by guohao0328** [comment]
如果不是使用的问题而是bug的话，我再等等新版本，目前不是很着急，谢谢

**Comment by guohao0328** [comment]
https://gitee.com/guohao0328/mock-data.git 这个是我的样例程序地址

**Comment by fvarrui** [comment]
Hi @guohao0328!
I've just tested your mock project, but I got another error:

![image](https://user-images.githubusercontent.com/4354486/212194669-a6b04c2f-5671-4bba-81f2-d8b98cf492eb.png)

This is a known issue related to modular JARs which can be fixed setting property `customizedJre=false`.

After testing again, mockdata app is built without problems, but I got this error when running the EXE:

![image](https://user-images.githubusercontent.com/4354486/212196150-2886fc08-9cf7-4862-a7a8-15415c22ff46.png)

> JavaFX runtime components are missing

This is another known problem, which can be fixed moving your `main()` method to another class (e.g. also called `Main`), like this:

```java
package org.guohao.mockdata;

import javafx.application.Application;

public class Main {
    
    public static void main(String[] args) {
        Application.launch(MockDataApplication.class, args);
    }

}
```

Finally, use this new class as your mainClass (`mainClass=org.guohao.mockdata.Main`), build again and ... voilà:

![image](https://user-images.githubusercontent.com/4354486/212196707-632af919-c0cc-4b0f-b2a4-5beeac391299.png)

Your app is running using the generated EXE file.
